lib/Twig/Template.php

```diff
@@ -244,6 +244,38 @@ public function hasBlock($name)
     }
 
     /**
+     * Returns whether a block exists or not in the current context of the template.
+     *
+     * This method is for internal use only and should never be called
+     * directly.
+     *
+     * This method does return blocks defined in the current template
+     * or defined in "used" traits or defined in parent templates.
+     *
+     * @param string $name    The block name
+     * @param array  $context The context
+     * @param array  $blocks  The current set of blocks
+     *
+     * @return bool true if the block exists, false otherwise
+     */
+    public function existsBlock($name, array $context, array $blocks = array())
```

Comment:
Can you also add an `@internal` tag like I've just done in #1835 for other internal methods?